src/main/java/codechicken/nei/api/API.java

```diff
@@ -310,8 +310,34 @@ public static void addRecipeCatalyst(ItemStack stack, Class<? extends IRecipeHan
      * @param stacks the ingredients that can craft recipes (like a furnace or crafting table)
      * @param handler the recipe category handled by the ingredient
      */
-    public static void addRecipeCatalyst(List<ItemStack> stacks, Class<? extends IRecipeHandler> handler) {
-        RecipeCatalysts.addRecipeCatalyst(stacks, handler);
+    public static void addRecipeCatalyst(List<ItemStack> stacks, IRecipeHandler handler) {
+        String handlerID = handler.getOverlayIdentifier();
+        if (handlerID != null) {
+            addRecipeCatalyst(stacks, handler.getOverlayIdentifier());
+        }
+        else {
+            NEIClientConfig.logger.warn(String.format("failed to add recipe catalyst, implement IRecipeHandler#getOverlayIdentifier for your handler %s", handler.getClass().getName()));
```

Comment:
Is this fatal? Or do things "keep working" without this?

Reply:
Wdym by "working"? It will just skip that invalid command and other catalysts or functions will work just fine.
And since I'm going to work on all mods that register handlers (tedious!), I can fill in that method to make it working.
Plus I can see many mods already implement it.

Reply:
Sounds good, just didn't want uncatalysted recipes to start disappearing.